File: leetcode/Hot-100/c++/hmjl.cpp
class Solution {
public:
    int hammingDistance(int x, int y) {
        // x 和 y 先进行异或操作 然后不断右移一位，计算1的个数
        int s = x ^ y, ret = 0;
        while (s) {
            // 按为与操作 1&0=0 1&1=1
            ret += s & 1;
            // 右移一位  最左补0
            s >>= 1;
        }
        return ret;
    }
};

File: leetcode/Hot-100/python/hmjl.py
# -*- coding: utf-8 -*-
"""
两个整数之间的 汉明距离 指的是这两个数字对应二进制位不同的位置的数目。
给你两个整数 x 和 y，计算并返回它们之间的汉明距离。
例如：
输入：x = 1, y = 4
输出：2
解释：
1   (0 0 0 1)
4   (0 1 0 0)
"""
class Solution:
    def hammingDistance(self, x, y):
        """
        对两个字符串进行异或运算，并统计结果为1的个数，那么这个数就是汉明距离
        bin() 返回一个整数 int 或者长整数 long int 的二进制表示; count() 方法用于统计字符串里某个字符出现的次数
        """
        return bin(x^y).count("1")